packages/terra-form-select/src/shared/_Option.jsx

```diff
@@ -94,6 +94,12 @@ const Option = ({
     }
   }
 
+  let label = display;
+  if (customProps.index && customProps.totalOptions) {
+    label = `${display} (${customProps.index} ${customProps.ofText} ${customProps.totalOptions})`;
+  }
+  const itemLabel = isSelected || customProps.index === 1 ? `${customProps.expandedStateText} ${label}` : label;
+
```

Comment:
Let's create a private prop definitions for each of these props since they are known props and being passed down menu.jsx it's better to have prop declared for each of the above values(index, ofText, totalOptions and expandedStateText).
we need to ensure customProps should contain only props passed by consumer for <SingleSelect.Option> .

Including this props in customProps will add them to dom as well as shown in below screenshot
( all the above attributes are not required to be included on `li` dom) 
<img width="1339" alt="Screenshot 2023-01-19 at 11 27 56 PM" src="https://user-images.githubusercontent.com/12869809/213526467-96fc678c-4263-48d2-a0fc-2c7379a63ae1.png">

Reply:
Updated to private props in https://github.com/cerner/terra-core/pull/3707/commits/81fc0fed4e8d45e376fbb56046a06dc142177cf8